pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -78,6 +82,23 @@ type ApplicationStatus struct {
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 }
 
+// ImageSpec contains information about an image used as an icon.
+type ImageSpec struct {
+
+	// The source for image represented as either an absolute URL to the image or a Data URL containing
+	// the image. Data URLs are defined in RFC 2397.
+	Source string `json:"src"`
+
+	// (optional) The height of the image in pixels.
+	Height string `json:"height,omitempty"`
+
+	// (optional) The width of the image in pixels.
+	Width string `json:"width,omitempty"`
+
+	// The mine type of the image (e.g., "image/png").
+	Type string `json:"type,omitempty"`
```

Comment:
is this optional?

Reply:
Data URLs are going to contain it, so it may be unnecessary in that case.

Reply:
Added the option note to the comment.